File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-5.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        return count(root, root.val);
    }
    public int count(TreeNode root, int maxtillnow) {
        if(root==null) return 0;
        int ret = root.val >= maxtillnow? 1 : 0;
        maxtillnow = Math.max(root.val, maxtillnow);
        ret += count(root.left, maxtillnow);
        ret += count(root.right, maxtillnow);
        return ret;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-5.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        idx = 0
        ans = -1
        def traverse(node: Optional[TreeNode]):
            nonlocal idx, ans
            if not node or ans!=-1:
                return
            traverse(node.left)
            idx += 1
            if idx==k:
                ans = node.val
                return
            traverse(node.right)
        traverse(root)
        return ans